### PR TITLE
Added Smithy Lexer (#1878)

### PR DESCRIPTION
**Pull request by immanuelqrw**
* Added Smithy Lexer
* Added Smithy Lexer author
* Documented Smithy as a supported language
* Added Smithy test file and output

### CODE REVIEW

File: pygments/lexers/smithy.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""
+Smithy IDL lexer
+~~~~~~~~~~~~~~~~
+
+Lexers for the Smithy IDL.
+"""
```
Comment (Anteru) [inline pygments/lexers/smithy.py:7]: Missing the license blurb. All Pygments contributions are under BSD and the file header should mention that.

File: pygments/lexers/smithy.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+from pygments.lexer import RegexLexer, default, include, bygroups
+from pygments.token import Text, Comment, Operator, Keyword, Name, String, \
+    Number, Literal
+
+__all__ = ['SmithyLexer']
+
+
+class SmithyLexer(RegexLexer):
+    """
+    For Smithy IDL
```
Comment (Anteru) [inline pygments/lexers/smithy.py:20]: Please add `.. versionadded: 2.10`

File: pygments/lexers/smithy.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            (r'\[', Text, "#push"),
+            (r'\]', Text, "#pop"),
+            (r'\(', Text, "#push"),
+            (r'\)', Text, "#pop"),
+            (r'{', Text, "#push"),
+            (r'}', Text, "#pop"),
+            (r'"{3}(\\\\|\n|\\")*"{3}', String.Doc),
+            (r'"(\\\\|\n|\\"|[^"])*"', String.Double),
+            (r"'(\\\\|\n|\\'|[^'])*'", String.Single),
+            (r'[:,\s]+', Text),
```
Comment (Anteru) [inline pygments/lexers/smithy.py:58]: Can you please split this rule into `:,` -> Punctuation and `\s` -> Text.Whitespace? We're trying to be a bit more consistent on whitespace handling.

File: pygments/lexers/smithy.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            (r'///.*$', Comment.Multiline),
+            (r'//.*$', Comment),
+            (r'@[0-9a-zA-Z\.#-]*', Name.Decorator),
+            (r'(=)', Name.Decorator),
+            (r'^(\$version)(:)(.+)', bygroups(Keyword.Declaration, Name.Decorator, Name.Class)),
+            (r'^(namespace)(\s+' + identifier + r')\b', bygroups(Keyword.Declaration, Name.Class)),
+            (r'^(use|byte|short|integer|long|float|'
+             r'document|double|bigInteger|bigDecimal|boolean|blob|string|timestamp)(\s+' + identifier + r')\b',
+             bygroups(Keyword.Declaration, Name.Class)),
+            (r'^(apply|list|map|set|structure|union|resource|operation|service|trait)(\s+' + identifier + r')',
```
Comment (Anteru) [inline pygments/lexers/smithy.py:41]: We have a `words()` function to efficiently match lists of words, can you please change those two to use that?

**Review by Anteru — CHANGES_REQUESTED** [review]

File: pygments/lexers/smithy.py (hunk 1 above)
Comment (immanuelqrw) [reply pygments/lexers/smithy.py]: Added.

File: pygments/lexers/smithy.py (hunk 2 above)
Comment (immanuelqrw) [reply pygments/lexers/smithy.py:22]: Added.

File: pygments/lexers/smithy.py (hunk 3 above)
Comment (immanuelqrw) [reply pygments/lexers/smithy.py]: Sure, makes sense. Added.

File: pygments/lexers/smithy.py (hunk 4 above)
Comment (immanuelqrw) [reply pygments/lexers/smithy.py]: Ah cool, added.

**Comment by Anteru** [comment]
Thanks for the timely updates!

**Comment by immanuelqrw** [comment]
Forgot to re-run `_mapping.py` after the last changes. Ran that and committed the updated aliases and mappings.

**Comment by immanuelqrw** [comment]
Not sure why `pypy3 · ubuntu-latest` test failed.

`test_happy_javascript_fragment` failed, but no changes were made that should have affected that.

**Error**:
```
  >       assert time.time() - start_time < MAX_HL_TIME, \
              'The HTML lexer might have an expensive happy-path script case'
  E       AssertionError: The HTML lexer might have an expensive happy-path script case
  E       assert (1628191328.1567895 - 1628191316.165653) < 10
  E        +  where 1628191328.1567895 = <built-in function time>()
  E        +    where <built-in function time> = time.time
  E        +    where <built-in function time> = time.time
  
  /home/runner/work/pygments/pygments/tests/test_html_lexer.py:31: AssertionError
```

Seems like a randomly slow test event?
Wasn't sure if there was something I could do to look into this, or if it's just a flaky test that needs to be re-run.

**Comment by Anteru** [comment]
Yes, that's a slow test failing once in a while. Don't worry :)

**Comment by Anteru** [comment]
Merged, thanks for your contribution!

**Pull request merged**